src/pocketmine/Player.php

```diff
@@ -2241,7 +2242,34 @@ public function handleInventoryTransaction(InventoryTransactionPacket $packet) :
 
 				switch($type){
 					case InventoryTransactionPacket::USE_ITEM_ON_ENTITY_ACTION_INTERACT:
-						break; //TODO
+						if(!$target->isAlive()){
+							return true;
+						}
+
+						$heldItem = $this->inventory->getItemInHand();
+
+						$ev = new PlayerGentlyTouchEntityEvent($this, $target, $heldItem, $packet->trData->clickPos);
+
+						if(!$this->canInteract($target, 8)){
+							$ev->setCancelled();
+						}
+
+						$this->server->getPluginManager()->callEvent($ev);
+
+						$target->interact($ev);
+
+						if(!$ev->isCancelled()){
```

Comment:
Seems to be an inconsistency here. The item is only receiving the click event if the event is not cancelled, but the entity will always receive it, cancelled or not.

Additionally, parameter inconsistency exists here.

Reply:
For Entity::interact() I tried to be consistent with other methods in Entity ( Entity::attack ). 
For Item::onClickEntity, I tried to be consistent with other methods in Item ( Item::onClickAir, Item::onActivate ).